src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts

```diff
@@ -1,12 +1,399 @@
 export const description = `
-TODO:
-- Test pipeline outputs with different color targets, depth/stencil targets.
-  - different scalar types in shader (f32, u32, i32) to targets with different format (f32 to unorm/float, u32 to uint)
-  - different componentCounts of the output doesn't matter (e.g. f32, vec2<f32>, vec3<f32>, vec4<f32>)
-    Extra components are discarded and missing components are filled to 0,0,1.
+- Test pipeline outputs with different color target formats.
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { unreachable } from '../../../../common/util/util.js';
+import { kRenderableColorTextureFormats, kTextureFormatInfo } from '../../../capability_info.js';
 import { GPUTest } from '../../../gpu_test.js';
 
-export const g = makeTestGroup(GPUTest);
+class F extends GPUTest {
+  getExpectedTypeAndComponentCount(
```

Comment:
Add `sampleType` to table with `GPUTextureSampleType` values (`float`, `sint`, `uint`, `depth`) to avoid further confusion. (Formatting with full words doesn't turn out to be a problem)